library/src/main/java/com/getstream/sdk/chat/utils/UploadManager.java

```diff
@@ -0,0 +1,125 @@
+        if (isMedia && attachment.getType().equals(ModelType.attach_image))
+            channel.sendImage(attachment.config.getFilePath(), "image/jpeg", callback);
+        else
+            channel.sendFile(attachment.config.getFilePath(), attachment.getMime_type(), callback);
+    }
+
+    private UploadFileCallback getUploadFileCallBack(Attachment attachment,
+                                                     boolean isMedia,
+                                                     UploadFileListener fileListener) {
+        return new UploadFileCallback<UploadFileResponse, Integer>() {
+            @Override
+            public void onSuccess(UploadFileResponse response) {
+                fileUploadSuccess(attachment, response, isMedia, fileListener);
+            }
+
+            @Override
+            public void onError(String errMsg, int errCode) {
+                fileUploadFailed(attachment, errMsg, fileListener);
+            }
+
+            @Override
+            public void onProgress(Integer percentage) {
+                fileUploading(attachment, percentage, fileListener);
+            }
+        };
+    }
+
+    private void fileUploadSuccess(Attachment attachment,
+                                   UploadFileResponse response,
+                                   boolean isMedia,
+                                   UploadFileListener fileListener) {
+        updateQueue(attachment,false);
+        if (!attachment.config.isSelected())
+            return;
+
+        if (isMedia && attachment.getType().equals(ModelType.attach_image)) {
+            File file = new File(attachment.config.getFilePath());
+            attachment.setImageURL(response.getFileUrl());
+            attachment.setFallback(file.getName());
+        } else {
+            attachment.setAssetURL(response.getFileUrl());
+        }
+
+        attachment.config.setUploaded(true);
+        fileListener.onSuccess(attachment);
+
+    }
+
+    private void fileUploadFailed(Attachment attachment,
+                                  String errMsg,
+                                  UploadFileListener fileListener) {
+        Utils.showMessage(context, errMsg);
+        fileListener.onFailed(attachment);
+    }
+
+    private void fileUploading(Attachment attachment,
+                               Integer percentage,
+                               UploadFileListener fileListener) {
+        attachment.config.setProgress(percentage);
+        fileListener.onProgress(attachment);
+    }
+
+    public void updateQueue(Attachment attachment, boolean add){
+        if (queue == null)
+            queue = new ArrayList<>();
+        if (add){
+            queue.add(attachment);
+        }else {
+            queue.remove(attachment);
+        }
+    }
+
+    public boolean isUploadingFile() {
+        return queue != null && !queue.isEmpty();
+    }
+
+    public void setQueue(List<Attachment> queue) {
```

Comment:
If it's used only for clearing queue I would not allow to pass anything externally, but replace it actual method:
```java
public void reset() {
  queue.clear();
  //maybe callbacks that notifies that there is no uploading processes
}
```